src/SFA.DAS.AdminService.Web.Tests/Controllers/Roatp/DownloadRoatpControllerTests.cs

```diff
@@ -0,0 +1,64 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.Extensions.Logging;
+using Moq;
+using NUnit.Framework;
+using SFA.DAS.AdminService.Web.Controllers.Roatp;
+using SFA.DAS.AdminService.Web.Helpers;
+using SFA.DAS.AdminService.Web.Infrastructure.RoatpClients;
+using SFA.DAS.AdminService.Web.Models.Roatp;
+using SFA.DAS.AdminService.Web.Services;
+using SFA.DAS.AdminService.Web.ViewModels.Roatp;
+using SFA.DAS.AssessorService.ApplyTypes.Roatp;
+
+namespace SFA.DAS.AdminService.Web.Tests.Controllers.Roatp
+{
+    [TestFixture]
+    public class DownloadRoatpControllerTests
+    {
+        private DownloadRoatpController _controller;
+        private Mock<IRoatpApplicationApiClient> _applicationApplyApiClient;
+        private Mock<ICsvExportService> _csvExportService;
+
+        [SetUp]
+        public void SetUp()
+        {
+            _applicationApplyApiClient = new Mock<IRoatpApplicationApiClient>();
+            _csvExportService = new Mock<ICsvExportService>();
+
+            _controller = new DownloadRoatpController(Mock.Of<IRoatpApiClient>(), 
+                Mock.Of<IDataTableHelper>(), 
+                Mock.Of<ILogger<DownloadRoatpController>>(),
+                _csvExportService.Object, 
+                _applicationApplyApiClient.Object
+            );
+        }
+
+        [Test]
+        public async Task ApplicationDownloadCsv_downloads_file()
+        {
+            var apiResponse = new List<RoatpApplicationOversightDownloadItem>();
+
+            _applicationApplyApiClient.Setup(
+                x => x.GetApplicationOversightDetailsForDownload(It.IsAny<DateTime>(), It.IsAny<DateTime>()))
+                .ReturnsAsync(apiResponse);
+
+            var expectedFileContents = Encoding.ASCII.GetBytes("THIS IS A TEST");
```

Comment:
This is not valid CSV content, thus not a realistic test scenario

Reply:
The test is to verify that the output from the csv export service is what is returned from the action, not that the content is in a CSV format (which is a CSVHelper concern), so the test is valid.

Reply:
True, although it's better if the file contents reflected reality. Otherwise why go to the efforts of initializing & validating the contents below.

A better test would be to check that the result isn't null